### PR TITLE
Add port binding rally workload file

### PR DESCRIPTION
**Pull request by huikang**
- register docker sandbox in rally
- Working port bind
- Add create and list port workload file

Signed-off-by: Hui Kang kangh@us.ibm.com

### CODE REVIEW

**Comment by huikang** [comment]
Port binding workload is done. Now we can stress the southbound database and the ovn-controller. - Hui

**Comment by l8huang** [comment]
Seems like 'install_method' is a problem now, it is coming up everywhere :)

'docker' and 'bare-metal' are different deployment methods in my mind, why not move it into deployment config, e.g. add a 'method'  to [ovn-multihost-deployment.json.j2](https://github.com/openvswitch/ovn-scale-test/blob/master/ansible/roles/rally/templates/ovn-multihost-deployment.json.j2#L7), record it in Deployment object in DB, then initialize ovsclients with the deployment 'method' at [rally_ovs.plugins.ovs.scenario.OvsScenario](https://github.com/openvswitch/ovn-scale-test/blob/master/rally_ovs/plugins/ovs/scenario.py#L38). In this way, the 'install_method' is only set once while deploying, all of tasks can follow it without additional configuration. 

In Rally's way, 'install_method' of 'docker' is better to be done with a deployment engine plugin, like 'OvnSandboxControllerEngine' and 'OvnSandboxFarmEngine'. You can wrap ansible commands mentioned in [README:Deploying OVN Emulation](https://github.com/huikang/ovn-scale-test/blob/ansible-docker-rally-port-binding/ansible/README.rst#deploying-ovn-emulation)  and [README:Registerring with Rally](https://github.com/huikang/ovn-scale-test/blob/ansible-docker-rally-port-binding/ansible/README.rst#registerring-with-rally) into the engine plugin.

File: ansible/roles/rally/templates/create_and_bind_ports.json.j2 [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
```
Comment (l8huang) [inline ansible/roles/rally/templates/create_and_bind_ports.json.j2]: It's better to keep these task config files under [samples](https://github.com/huikang/ovn-scale-test/tree/ansible-docker-rally-port-binding/samples) folder.

**Comment by huikang** [comment]
Thanks for the suggestions. I think it would be better to split this patch into two smaller ones: one for the install_method and the other for the port_binding. I will create another PR and leave this as a reference.

**Comment by huikang** [comment]
Lei, I was trying to add install_method as a [field to the deployment database](https://github.com/huikang/ovn-scale-test/blob/change-install-method/rally_ovs/plugins/ovs/deployment/engines/ovn_sandbox_controller.py#L133) as you suggested.

Later when creating the OvsClient, we can use `self._controller_clients = ovsclients.Clients(cred, install_method)`

My question is how can I retrieve this install_method when creating the [ovsclient](https://github.com/openvswitch/ovn-scale-test/blob/master/rally_ovs/plugins/ovs/scenario.py#L38).

Something like cred = v["install_method"] gives me compile error.

Thanks.

**Comment by l8huang** [comment]
Deployment object has a field [config](https://github.com/openstack/rally/blob/master/rally/common/db/sqlalchemy/models.py#L80), it contains the config file's content used to create the deployment, so you can retrieve install method from it.

**Comment by huikang** [comment]
To add `install_method` field into the deployment object, do I need to call `add_resource` like you did for [resourcetype.CONTROLLER](https://github.com/openvswitch/ovn-scale-test/blob/master/rally_ovs/plugins/ovs/deployment/engines/ovn_sandbox_controller.py#L127)?

I think I figure it out: dep.config is an accessible list. Thanks for your suggestion and will update a new PR.

**Comment by huikang** [comment]
Merged as smaller PRs

**Pull request closed without merging**